.pytool/Plugin/EccCheck/EccCheck.py

```diff
@@ -138,11 +138,23 @@ def RunBuildPlugin(self, packagename, Edk2pathObj, pkgconfig, environment, PLM,
             return 1
 
     def GetDiff(self, pkg: str, temp_diff_output: str) -> List[str]:
+        # MU_CHANGE [BEGIN]
+        #
+        # Run git command to determine repos default branch
+        #
+        result = StringIO()
+        params = "symbolic-ref refs/remotes/origin/HEAD --short"

```

Comment:
I agree the current way of hardcoding to origin/master is bad for us, but I think this way will also cause issues if we are targeting the non-default branch on CI. Such as cherry-picking changes back or if you have a PR targeting a feature branch.

Reply:
I agree that this tool is not optimal. Thankfully, this is not being enabled as part of CI, its scope is just being changed to `cibuild-edk2` to enable mu based projects to be able to run the tool locally. 

Comparing against the default branch may seem problematic, but the way the tool works is it uses the lines changed between the default branch and the current branch and filters the ecc check log to only report the errors corresponding to the changed lines. This means that the tool will only ever check a subset of the entire files for errors, and that a lot of things which do not conform to the coding standards will be able to slip through repeatedly until someone changes something in those lines.

Reply:
I think Joey's saying that we often work on non-default branches. For example, release/202311 (default) and release/202302. In that case, cherry-picking into release/202302 will have the files compared to their equivalents in release/202311 which might be in a different base state which would result in an invalid delta comparison.

The patchset only diff is okay I think but it needs to be correct.